server_tests/test/clientTests.js

```diff
@@ -175,438 +173,401 @@ describe("RUST API TEST SUITE", function() {
         });
 
-        it("Expect success. camera.startSession will timeout after the the specified timeout value", function(done) {
+        it("Expect success. camera.startSession will timeout after the the specified timeout value", function() {
             this.timeout(timeoutValue);
-            testClient.startSession(5)
-            .then( Comparison.catchExceptions(done, function(res) {
+            return testClient.startSession(5)
+            .then( function onSuccess (res) {
                 sessionId = res.body.results.sessionId;
                 Comparison.oscSessionOpOutput(res, {'sessionId': sessionId, 'timeout': 5});
                 return Q.delay(8000);
-            }))
+            }, shouldNotGetHere)
             .then( function() {
                 return testClient.startSession();
             })
-            .then( Comparison.catchExceptions(done, function(res) {
+            .then( function onSuccess (res) {
                 sessionId = res.body.results.sessionId;
                 Comparison.oscSessionOpOutput(res, {'sessionId': sessionId});
-                done();
-            }));
+            }, shouldNotGetHere);
         });
-            
-        it("Expect cameraInExclusiveUse Error. camera.startSession cannot start session while another session is already running", function(done) {
-            testClient.startSession()
-            .then( Comparison.catchExceptions(done, function(res) {
+
+        it("Expect cameraInExclusiveUse Error. camera.startSession cannot start session while another session is already running", function() {
+            return testClient.startSession()
+            .then(
+              function onSuccess (res) {
                 sessionId = res.body.results.sessionId;
                 Comparison.oscSessionOpOutput(res, {'sessionId': sessionId});
                 return testClient.startSession();
-            }))
-            .then( Comparison.catchExceptions(done, function(res) {
-                Comparison.cameraInExclusiveUseError(res);
-                done();
-            }));
-        });
-
-        it("Expect invalidParameterValue Error. camera.startSession cannot start session when incorrect timeout type is provided", function(done) {
-            testClient.startSession('wrongtype')
-            .then( Comparison.catchExceptions(done, function(res) {
-                Comparison.invalidParameterValueError(res);
-                done();
-            }));
+              },shouldNotGetHere)
+              .then(shouldNotGetHere,
+              function onError (err) {
+                  Comparison.cameraInExclusiveUseError(err);
+              }
+            );
+        });
+
+        it("Expect invalidParameterValue Error. camera.startSession cannot start session when incorrect timeout type is provided", function() {
+            return testClient.startSession('wrongtype')
+            .then( function onSuccess (res) {
+                assert.fail("Expected failure");
```

Comment:
In this case we're using `assert.fail` rather than `shouldNotGetHere`, which is used elsewhere for the same purpose. Be consistent.